Sources/Socket.swift

```diff
@@ -2423,9 +2441,27 @@ public class Socket: SocketReader, SocketWriter {
 			
 			// Is the caller's buffer big enough?
 			if bufSize < self.readStorage.length {
-				
-				// Nope, throw an exception telling the caller how big the buffer must be...
-				throw Error(bufferSize: self.readStorage.length)
+
+				// It isn't should we just use the available space?
+				if truncate {
+
+					// Yep, copy what storage we can and remove the bytes from the internal buffer.
+					memcpy(buffer, self.readStorage.bytes, bufSize)
+
+					#if os(Linux)
+						// Workaround for apparent bug in NSMutableData
+						self.readStorage = NSMutableData(bytes: self.readStorage.bytes.advanced(by: bufSize), length:self.readStorage.length - bufSize)
+					#else
+						self.readStorage.replaceBytes(in: NSRange(location:0, length:bufSize), withBytes: nil, length: 0)
+					#endif
+
+					return bufSize
+
+				} else {
+
+					// Nope, throw an exception telling the caller how big the buffer must be...
+					throw Error(bufferSize: self.readStorage.length)
+				}
```

Comment:
Good point. It crops up twice right here in this method which made it stand out to me, but especially after removing `truncate` from the API for UDP a dedicated function is probably not worth it.